test/test_nestedtensor.py

```diff
@@ -6166,6 +6166,98 @@ def test_unbind_backward(self, device, dtype):
         expected_grad.unbind()[1].add_(1.0)
         torch._dynamo.disable(self.assertEqual)(nt.grad, expected_grad)
 
+    def test_masked_select(self, device):
+        offsets = torch.tensor([0, 2, 5, 6], device=device)
+        njt = nested_view_from_values_offsets(
+            values=torch.arange(6, device=device), offsets=offsets
+        )
+        mask = nested_view_from_values_offsets(
+            values=torch.tensor([False, True, False, False, True, True], device=device),
+            offsets=offsets,
+        )
+
+        masked_nt = njt.masked_select(mask)
+        self.assertEqual(njt.values(), torch.arange(6, device=device))
+        self.assertEqual(masked_nt.values(), torch.tensor([1, 4, 5], device=device))
+        self.assertEqual(
+            masked_nt.offsets(),
+            torch.tensor([0, 1, 2, 3]),
+        )
+        self.assertEqual(masked_nt.lengths(), None)
+
+        njt = nested_view_from_values_offsets(
+            values=torch.arange(6, device=device), offsets=offsets
+        )
+        mask = torch.tensor([True, False, True], device=device)
+
+        masked_nt = njt.masked_select(mask)
+        self.assertEqual(njt.values(), torch.arange(6, device=device))
+        self.assertEqual(masked_nt.values(), torch.tensor([0, 1, 5], device=device))
+        self.assertEqual(
+            masked_nt.offsets(),
+            torch.tensor([0, 2, 2, 3]),
+        )
+        self.assertEqual(masked_nt.lengths(), None)
+
+    @skipIfTorchDynamo(
+        "Fails on returning a fresh tensor but this is expected behavior"
+    )
```

Comment:
Could you check if this is fixed by rebase? The PR fixing creating NT in graph should be landed.

Reply:
Double checked, we can delete this now

Reply:
Perfect